chips/nrf52/src/spi.rs

```diff
@@ -202,10 +215,8 @@ impl From<Frequency> for u32 {
             Frequency::M8 => 8_000_000,
         }
     }
-}
 
-impl From<u32> for Frequency {
-    fn from(freq: u32) -> Frequency {
+    pub fn from_spi_rate(freq: u32) -> Frequency {
```

Comment:
this still seems error-prone but seems to only be used internally with validated input? Perhaps make it private or document it?

Reply:
I didn't really touch the existing methods because I don't want to break any existing changes. It always returns a `Frequency` for every input, namely one that is lower or equal to the requested one, except for the lowest (which may be higher than requested). I think this makes sense in this context. I'm not yet sure why the `Frequency` enum is public at all - making it private would limit those methods to be called inside this module anyways.

Reply:
Right, I got confused by `register_value` vs the `actual rate`